subsys/bluetooth/audio/ascs.c

```diff
@@ -1156,6 +1156,9 @@ static int ascs_ep_set_codec(struct bt_audio_ep *ep, uint8_t id, uint16_t cid,
 	bt_pacs_cap_foreach(ep->dir, codec_lookup_id, &lookup_data);
 
 	if (lookup_data.codec == NULL) {
+		BT_DBG("Codec with id %u for dir %u is not supported by our capabilities",
+			ep->dir, codec->id);
```

Comment:
debug parameters are in wrong order (id first)
also check in line 1149 looks a bit suspicious and "codec->id" debug may lead to NULL pointer deference (as line 1165 assumes codec can be NULL)

maybe check in 1149 should only validate ep pointer? and then rest of function should handle codec==NULL ?
anyway, something fishy is happening in this function :)

Reply:
Nice catch. 

w.r.t. the check, I agree. Perhaps we should just remove the check, as this is a static function, and is only called one place with the address of two variables, so neither will ever be NULL.

Reply:
That check is outside the scope of this PR though :)